pkg/api/norman/customization/cluster/validator.go

```diff
@@ -665,13 +666,23 @@ func validateGKENodePools(spec *v32.ClusterSpec) error {
 
 		version := np.Version
 		if version == nil || *version == "" {
-			errors = append(errors, fmt.Sprintf("node pool [%s] version cannot be empty", name))
+			versionErrors = append(versionErrors, fmt.Sprintf("node pool [%s] version cannot be empty", name))
 			continue
 		}
+
+		image := strings.ToLower(np.Config.ImageType)
+		// Windows images are WINDOWS_LTSC or WINDOWS_SAC. The cluster must have at least one non-Windows node pool.
+		if !strings.Contains(image, "windows") {
+			hasRequiredLinuxPool = true
+		}
 	}
 
-	if len(errors) != 0 {
-		return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf(strings.Join(errors, ";")))
+	if len(versionErrors) != 0 {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf(strings.Join(versionErrors, ";")))
+	}
+
+	if !hasRequiredLinuxPool {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("at least 1 Linux node pool is required"))
```

Comment:
Is there a reason for separating these errors? For example, would it be bad to keep the name of the variable as `errors` and just append all errors to the slice, check the length of this slice, and return the error if any error is present?

Reply:
That sounds better, I'll do that